Framework/DataObjects/inc/MantidDataObjects/WorkspaceCreation.h

```diff
@@ -158,13 +149,39 @@ std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
       ws = detail::createConcreteHelper<T>();
     }
   }
+  return ws;
+}
 
+MANTID_DATAOBJECTS_DLL void
+initializeFromParent(const API::MatrixWorkspace &parent,
+                     API::MatrixWorkspace &ws);
+}
+
+template <class T, class P, class IndexArg, class HistArg,
+          class = typename std::enable_if<
+              std::is_base_of<API::MatrixWorkspace, P>::value>::type>
+std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
+                          const HistArg &histArg) {
+  std::unique_ptr<T> ws = detail::createUninitialized<T, P>(parent);
   ws->initialize(indexArg, HistogramData::Histogram(histArg));
   detail::initializeFromParent(parent, *ws);
 
   return ws;
 }
 
+template <class T, class P, class IndexArg,
+          class = typename std::enable_if<
+              std::is_base_of<API::MatrixWorkspace, P>::value>::type>
+std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
+                          const HistogramData::Histogram &histogram) {
```

Comment:
Why this new variant? I think it will actually break things since there is code that creates workspaces based on a histogram including data (via the templated version above).